// @parcel/transformer-xml/lib/rss.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.comments = void 0;
exports.description = description;
exports.enclosure = enclosure;
exports.url = exports.link = void 0;

var _utils = require("./utils");

// Handlers for elements defined by the RSS spec.
// See https://validator.w3.org/feed/docs/rss2.html
const link = _utils.urlHandler;
exports.link = link;
const url = _utils.urlHandler;
exports.url = url;
const comments = _utils.urlHandler;
exports.comments = comments;

function enclosure(element, asset) {
  let url = element.getAttribute('url');

  if (url) {
    url = asset.addURLDependency(url, {});
    element.setAttribute('url', url);
  }
}

function description(element, asset, parts) {
  let parcelKey = `${asset.id}:${parts.length}`;
  asset.addDependency({
    specifier: parcelKey,
    specifierType: 'esm',
    bundleBehavior: 'inline'
  });
  parts.push({
    type: 'html',
    content: element.textContent,
    uniqueKey: parcelKey,
    bundleBehavior: 'inline'
  });
  let child = element.firstChild;

  while (child) {
    element.removeChild(child);
    child = child.nextSibling;
  }

  let el = element.ownerDocument.createElementNS('https://parceljs.org', 'inline');
  el.setAttribute('key', parcelKey);
  element.appendChild(el);
}
